service/history/queue/processing_queue.go

```diff
@@ -221,19 +221,25 @@ func (q *processingQueueImpl) Merge(
 		))
 	}
 
-	newQueueStates = append(newQueueStates, newProcessingQueueState(
+	overlappingQueueState := newProcessingQueueState(
 		q1.state.level,
 		maxTaskKey(q1.state.ackLevel, q2.state.ackLevel),
 		minTaskKey(q1.state.readLevel, q2.state.readLevel),
 		minTaskKey(q1.state.maxLevel, q2.state.maxLevel),
 		q1.state.domainFilter.Merge(q2.state.domainFilter),
-	))
+	)
+	if overlappingQueueState.readLevel.Less(overlappingQueueState.ackLevel) {
```

Comment:
should this handle during initializing the queue state?

Reply:
Updated. Moved the logic into queue state initialization.